core/capabilities/ccip/ccipevm/extraargscodec.go

```diff
@@ -0,0 +1,17 @@
+package ccipevm
+
+import (
+	"errors"
+
+	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+)
+
+type ExtraArgsCodec struct{}
+
+func NewExtraArgsCodec() ExtraArgsCodec {
+	return ExtraArgsCodec{}
+}
+
+func (ExtraArgsCodec) DecodeExtraData(extraArgs cciptypes.Bytes, sourceChainSelector cciptypes.ChainSelector) (map[string]any, error) {
+	return nil, errors.New("not implemented")
```

Comment:
Need to implement it if we're gonna merge it. Otherwise this line: https://github.com/smartcontractkit/chainlink-ccip/pull/455/files#diff-a16bcafc2684d6be248d96e8fdf5fa77ec115f311c0c73d13f3f942ad0edb0c4R365 will just error every time.

Reply:
Yes, returning empty should be ok, but we can't return error.

Reply:
This issue was fixed, @makramkd .
Since it is night time for you, and this PR is critical to merge because it fixes a breaking interface change on chainlink-ccip PR, I will dismiss your "Changes Requested".